### PR TITLE
Explicit registry flag feature

### PR DESCRIPTION
**Pull request by noherczeg**
Corrected version of #563

Again, sorry for messing up the previous PR.

### CODE REVIEW

File: src/commands/PublishCommand.js [hunk 1]
```diff
@@ -338,14 +338,15 @@ export default class PublishCommand extends Command {
     });
 
     this.progressBar.init(this.packagesToPublish.length);
+    this.npmRegistry = this.getOptions().registry;
```
Comment (evocateur) [inline src/commands/PublishCommand.js]: This should be done once, in the initialize method.

Comment (noherczeg) [inline src/commands/PublishCommand.js]: Moved to runPreparations() so every command has access to it. I hope it does not count as an overkill :)

**Comment by evocateur** [comment]
Sounds good!

**Comment by evocateur** [comment]
Once @doug-wade has a chance to review this, we'll be good to go.

**Comment by noherczeg** [comment]
Thanks @evocateur ! :)

**Comment by gigabo** [comment]
Looks like the master build [failed](https://travis-ci.org/lerna/lerna/jobs/202430796) after this landed.

image

**Comment by doug-wade** [comment]
ugh.  must have collided.  I'll hop on it.

**Pull request merged**